experimental/packages/opentelemetry-exporter-prometheus/src/PrometheusExporter.ts

```diff
@@ -137,7 +140,8 @@ export class PrometheusExporter extends MetricReader {
    * Starts the Prometheus export server
    */
   startServer(): Promise<void> {
-    return new Promise(resolve => {
+    return new Promise((resolve, reject) => {
+      this._server.on('error', reject);
```

Comment:
Does `error` mean 100% for sure that listen failed and `resolve` won't be called?

Reply:
I think so. From the [docs](https://node.readthedocs.io/en/latest/api/net/):

> When the ['connect'](https://node.readthedocs.io/en/latest/api/net/#net_event_connect) event is emitted the socket is established. If there is a problem connecting, the 'connect' event will not be emitted, the 'error' event will be emitted with the exception.
> ...
> The connectListener parameter will be added as a listener for the ['connect'](https://node.readthedocs.io/en/latest/api/net/#net_event_connect) event.

connectListener the second parameter that is passed to `connect()`. So either one of them is called in any case, but not both. 

I adapted this line to only listen using `once()` in [c87043f](https://github.com/open-telemetry/opentelemetry-js/pull/3552/commits/c87043fda1b8a05453d32ba0941c0e8e14d481ae) as keeping it around probably is not necessary (any further invocations of `resolve()` or `reject()` would be no-ops)